ratis-metrics/src/test/java/org/apache/ratis/metrics/TestHadoop2MetricsAdaptor.java

```diff
@@ -1,75 +0,0 @@
-/**
- * Licensed to the Apache Software Foundation (ASF) under one
- * or more contributor license agreements.  See the NOTICE file
- * distributed with this work for additional information
- * regarding copyright ownership.  The ASF licenses this file
- * to you under the Apache License, Version 2.0 (the
- * "License"); you may not use this file except in compliance
- * with the License.  You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-package org.apache.ratis.metrics;
-
-import static org.junit.Assert.assertTrue;
-
-import java.util.HashMap;
-import java.util.Map;
-import java.util.concurrent.TimeUnit;
-
-import com.codahale.metrics.Counter;
```

Comment:
Thanks the question. As it's discussed in the mailing list, the metrics system should be independent from the chosen reporter. I removed all the dependency to the hadoopmetric adaptor but you can choose _any_ adaptor.

In case of Hadoop metrics, you can use it in your own application with:

```
MetricRegistries registries = MetricRegistries.global();
registries.addReporterRegistration((registry) ->
        HadoopMetrics2Reporter.forRegistry(registry.getDropWizardMetricRegistry())
            .build(DefaultMetricsSystem.instance(),
                registry.getMetricRegistryInfo().getName(),
                registry.getMetricRegistryInfo().getMetricsDescription(),
                registry.getMetricRegistryInfo().getMetricsComponentName())
            .start(30, TimeUnit.SECONDS));

....

   RaftServer raftServer = RaftServer.newBuilder()
        .setServerId(RaftPeerId.valueOf(id))
        .setStateMachine(stateMachine).setProperties(properties)
        .setGroup(raftGroup)
        .build();

    raftServer.start();
```
(After adding `com.github.joshelser:dropwizard-metrics-hadoop-metrics2-reporter:0.1.2` dependency).

We can propose a helper function for this in `dropwizard-metrics-hadoop-metrics2-reporter` project and the untit tests can also be moved to there (as there is no more hadoop metrics specific thing in ratis).

This makes it easier to use ratis anywhere outside of the Hadoop world (or with different version of Hadoop) but provide the flexibility to use any reporter include this hadoop2adaptor.